python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -1149,3 +1149,30 @@ def evaluate_for_asset(self, context: RuleEvaluationContext) -> RuleEvaluationRe
         if rate_limited_asset_partitions:
             return [(None, rate_limited_asset_partitions)]
         return []
+
+
+@whitelist_for_serdes
+class BlockingAssetCheckRule(AutoMaterializeRule, NamedTuple("_BlockingAssetcheckRule", [])):
```

Comment:
naming:

we need to pick this name, a name for a method on `AutoMaterializeRule`, and name for a condition

How about `skip_on_blocked_by_asset_check` / `BlockedByAssetCheckRule` and `BlockedByAssetCheck` condition?

If we want to be more verbose,

`skip_on_failed_upstream_blocking_asset_check`/`FailedUpstreamBlockingAssetCheckRule`/`FailedUpstreamBlockingAssetCheck` condition